Linking CXX executable bin/tpp-opt failed

I am trying to build the project for the first time. Whilst doing so I run into a linking issue when it comes to producing `bin/tpp-opt`. The error could be a mistake from my side, but it might still be useful to discuss here in case someone else were to run into similar issues when starting out as well.

I have followed the instructions from the [current README](https://github.com/plaidml/tpp-mlir/blob/f7ea8fd05acbafbbba3ff840cd4fa16cc8c5124f/README.md). Building `llvm-project` and most of `tpp-mlir` goes well, i.e. running `cmake -G Ninja .. -DCMAKE_BUILD_TYPE=RelWithDebInfo -DMLIR_DIR=$CUSTOM_LLVM_ROOT/lib/cmake/mlir -DLLVM_EXTERNAL_LIT=$CUSTOM_LLVM_ROOT/bin/llvm-lit -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++` succeeds. 

Next, I run `cmake --build . --target check-tpp`. This succeeds for 216 targets (out of 219) and then crashes:
```
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
[217/219] Linking CXX executable bin/tpp-opt
```

The errors right above these lines are as follows (note that I am eliding the command itself as it fills many pages of my terminal):

```
/usr/bin/ld: lib/libTPPGPU.a(GpuPipeline.cpp.o): in function `(anonymous namespace)::GpuPipeline::constructPipeline()':
/home/rolf/intel_code/tpp-mlir/lib/TPP/GPU/GpuPipeline.cpp:150:(.text._ZN12_GLOBAL__N_111GpuPipeline17constructPipelineEv+0x24d): undefined reference to `mlir::tpp::createTileConsumerAndFuseProducers(mlir::tpp::TileConsumerAndFuseProducersOptions const&)'
/usr/bin/ld: lib/libTPPPassBundles.a(LinalgLowering.cpp.o): in function `LinalgLowering::constructPipeline()':
/home/rolf/intel_code/tpp-mlir/lib/TPP/PassBundles/LinalgLowering.cpp:51:(.text._ZN14LinalgLowering17constructPipelineEv[_ZN14LinalgLowering17constructPipelineEv]+0x59): undefined reference to `mlir::tpp::createCombineXsmmOpPass()'
/usr/bin/ld: lib/libTPPPassBundles.a(LowLevelParallelization.cpp.o): in function `LowLevelParallelization::constructPipeline()':
/home/rolf/intel_code/tpp-mlir/lib/TPP/PassBundles/LowLevelParallelization.cpp:67:(.text._ZN23LowLevelParallelization17constructPipelineEv[_ZN23LowLevelParallelization17constructPipelineEv]+0xeb): undefined reference to `mlir::tpp::createSCFParallelLoopTiling(mlir::tpp::SCFParallelLoopTilingOptions const&)'
/usr/bin/ld: /home/rolf/intel_code/tpp-mlir/lib/TPP/PassBundles/LowLevelParallelization.cpp:69:(.text._ZN23LowLevelParallelization17constructPipelineEv[_ZN23LowLevelParallelization17constructPipelineEv]+0x119): undefined reference to `mlir::tpp::createIntelAMXTileConfigInsertionPass()'
/usr/bin/ld: /home/rolf/intel_code/tpp-mlir/lib/TPP/PassBundles/LowLevelParallelization.cpp:73:(.text._ZN23LowLevelParallelization17constructPipelineEv[_ZN23LowLevelParallelization17constructPipelineEv]+0x2b5): undefined reference to `mlir::tpp::createIntelAMXTileConfigHoistingPass()'
/usr/bin/ld: lib/libTPPPassBundles.a(TppMapping.cpp.o): in function `TppMapping::constructPipeline()':
/home/rolf/intel_code/tpp-mlir/lib/TPP/PassBundles/TppMapping.cpp:56:(.text._ZN10TppMapping17constructPipelineEv[_ZN10TppMapping17constructPipelineEv]+0x31): undefined reference to `mlir::tpp::createConvInitSimplify()'
/usr/bin/ld: /home/rolf/intel_code/tpp-mlir/lib/TPP/PassBundles/TppMapping.cpp:62:(.text._ZN10TppMapping17constructPipelineEv[_ZN10TppMapping17constructPipelineEv]+0xfb): undefined reference to `mlir::tpp::createRewriteConvToMatmulOrBrgemm()'
/usr/bin/ld: /home/rolf/intel_code/tpp-mlir/lib/TPP/PassBundles/TppMapping.cpp:71:(.text._ZN10TppMapping17constructPipelineEv[_ZN10TppMapping17constructPipelineEv]+0x1bf): undefined reference to `mlir::tpp::createConstantFoldPack()'
/usr/bin/ld: /home/rolf/intel_code/tpp-mlir/lib/TPP/PassBundles/TppMapping.cpp:77:(.text._ZN10TppMapping17constructPipelineEv[_ZN10TppMapping17constructPipelineEv]+0x2bf): undefined reference to `mlir::tpp::createLinalgConvertCompareSelectToMaximumfPass()'
/usr/bin/ld: /home/rolf/intel_code/tpp-mlir/lib/TPP/PassBundles/TppMapping.cpp:79:(.text._ZN10TppMapping17constructPipelineEv[_ZN10TppMapping17constructPipelineEv]+0x332): undefined reference to `mlir::tpp::createTileConsumerAndFuseProducers()'
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
[217/219] Linking CXX executable bin/tpp-opt
ninja: build stopped: subcommand failed.
```

I can try to debug myself, though I figured those who are familiar with the codebase might have a better clue as to what might be wrong/what I did wrong.

Any tips as to what to check are appreciated!

--- a/README.md
+++ b/README.md
@@ -86,7 +86,8 @@ cmake -G Ninja .. \
    -DMLIR_DIR=$CUSTOM_LLVM_ROOT/lib/cmake/mlir \
    -DLLVM_EXTERNAL_LIT=$CUSTOM_LLVM_ROOT/bin/llvm-lit \
    -DCMAKE_C_COMPILER=clang \
-   -DCMAKE_CXX_COMPILER=clang++ 
+   -DCMAKE_CXX_COMPILER=clang++ \
+   -DLLVM_USE_LINKER=lld
 cmake --build . --target check-tpp
 
 popd

